app/models/proposal.rb

```diff
@@ -115,14 +115,18 @@ def root_approval=(root)
       approval.set_list_position(idx + 1)   # start with 1
     end
 
+    self.clean_up_old_approvals(old_approvals, approval_list)
+
+    root.initialize!
+    self.reset_status()
+  end
+
+  def clean_up_old_approvals(old_approvals, approval_list)
     old_approvals.each do |old|
       unless approval_list.include?(old)
-        old.destroy()
+        old.destroy() if Approval.exists?(old.id)
       end
     end
```

Comment:
Minor: this could be simplified to

```ruby
approvals_to_remove = old_approvals - approval_list
approvals_to_remove.each(&:destroy!)
```

_Not that much simpler, now that I look at it. Up to you._